### PR TITLE
Everywhere: Fix all the typos

### PR DESCRIPTION
**Pull request by BenWiederhake**
These typos were found by running:

```
git ls-files -z '*.md' | xargs -0 cat Base/home/anon/Documents/tips.txt | aspell --lang=en_US list --ignore=3 | sort -u > words_flagged.lst
join --nocheck-order -v1 words_flagged.lst words_allow_extra.lst | tee words_misspelled.lst
```

… and then hand-filtering all the words that aren't known to aspell. Most notably:
- Due to sheer volume, I skipped all words with three or fewer letters.
- Due to sheer volume, I only searched for misspells in `.md` files (but fixed any finds in all files)
- I may have accidentally accepted a misspelling in this list of 2443 "new" words, which you can inspect here: https://gist.github.com/BenWiederhake/6d118d8fe8f66d1515d17bf9fa65532a
- I changed "neighbour" to "neighbor" and "behaviour" to "behavior", given that these are by far the most popular spellings in the project, and that the project's language is American English.

My favorite typo is either "gouped" or "LADBIRD", I can't decide :D

### CODE REVIEW

**Comment by MacDue** [comment]
Is behaviour to behavior, or neighbour to neighbor and such really a typo fix? Especially when some of these are in spec comments? Also the versions that lack the `u` look very wrong to me.

**Comment by Hendiadyoin1** [comment]
I agree with MacDue, spec comments should use the spelling that the spec uses

**Comment by BenWiederhake** [comment]
These concerns already are addressed.
- Yes, the difference between `neighbour` and `neighbor` exists. One is British English, the other is American English, and Serenity [explicitly chooses American English](https://github.com/SerenityOS/serenity/blob/5d2e9156239cd707a22ecea6c87d48e5fc1cbe84/CONTRIBUTING.md?plain=1#L30). So let's use only one spelling, instead of random unpredictable spelling.
- As the PR description already says, the PR doesn't change the spelling where a spec is involved. So I already follow your suggestion.

File: Userland/Libraries/LibGfx/ICC/Profile.cpp [hunk 1]
```diff
@@ -1573,7 +1573,7 @@ ErrorOr<void> Profile::from_pcs(FloatVector3 const& pcs, Bytes color) const
             //
             // where redTRC^-1, greenTRC^-1, and blueTRC^-1 indicate the inverse functions of the redTRC greenTRC and
             // blueTRC functions respectively.
-            // If the redTRC, greenTRC, or blueTRC function is not invertible the behaviour of the corresponding redTRC^-1,
+            // If the redTRC, greenTRC, or blueTRC function is not invertible the behavior of the corresponding redTRC^-1,
```
Comment (MacDue) [inline Userland/Libraries/LibGfx/ICC/Profile.cpp:1576]: This is a spec comment (and a few other things look like spec comments).
The extract line 'behaviour' exists in: https://www.color.org/specification/icc1v43_2010-12.pdf

Comment (MacDue) [reply Userland/Libraries/LibGfx/ICC/Profile.cpp:1576]: (That's why I mentioned spec comments)

Comment (BenWiederhake) [reply Userland/Libraries/LibGfx/ICC/Profile.cpp:1576]: Oh! I thought you meant the JS spec. Yup, good point, I'll revert that.

**Comment by BenWiederhake** [comment]
Changes since initial push:
- Dropped changes to the spec text in `LibGfx/ICC/` as suggested by @MacDue, thanks!
- Rebased for convenience.

**Review by GMTA — APPROVED** [review]

**Comment by trflynn89** [comment]
> As the PR description already says, the PR doesn't change the spelling where a spec is involved.

This is post-mortem, but just pointing out that this changed at least one spec text comment in LibJS/Bytecode/ASTCodegen.cpp. That comment was step 11 of https://tc39.es/ecma262/#sec-globaldeclarationinstantiation, so that should be reverted.

**Pull request merged**